### ISSUE TITLE
Bounty hunter missions are not removed as soon as the mission is completed (only after landing)

### ISSUE THREAD

**Issue by brendanjones** [endless-sky/endless-sky#9993]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Bounty hunter missions stay in your mission list even after you've destroyed the ships.

### Steps to Reproduce

1. Currently have any bounty hunting mission that requires destroying pirates/marauders
2. Destroy the ships
3. You get shown the completion message "The [ship name] has been eliminated. [...]"
4. Go to your missions
5. The mission is still there
6. Jump to another system, the mission is still there
7. Land on any planet, and the mission will disappear

### Expected Behavior

The mission would be removed from your mission list when step 3 occurs:

> 3. You get shown the completion message "The [ship name] has been eliminated. [...]"

### Screenshots

This should remove the mission:
<img width="351" alt="Screenshot 2024-04-15 at 12 32 43" src="https://github.com/endless-sky/endless-sky/assets/16049594/d98f5202-2d90-4a5b-bc1e-cf1687d7cdf8">

### Link to save file

Any pilot with enough combat rating to accept bounty hunter missions.

### Operating System

MacOS

### Game Source

Steam

### Game Version

0.10.6

**Comment by Anarchist2** [comment]
This is a consequence of how the bounty missions are implemented right now. Because missions can only be completed by landing at a specific destination planet, the bounty missions sidestep this by having the mission fail when the target ship is destroyed. Failed missions are only removed once you land because ships spawned by failed missions stick around until after you land. This could potentially be solved if it becomes possible to complete missions mid-flight, but that isn't a high priority at the moment.

**Comment by mOctave** [comment]
Well, technically speaking you get paid before you fail, I believe. But, yes, that's more or less how it works. There are lots of missions that you "fail" while completing properly. Although, come to think of it it might be a good idea to add a `"Bounty Hunting (X): done" ++` to all those bounty jobs.

**Comment by OcelotWalrus** [comment]
Personally, I don't think that's a bug, or at least a problem.

**Comment by Hurleveur** [comment]
> Well, technically speaking you get paid before you fail, I believe. But, yes, that's more or less how it works. There are lots of missions that you "fail" while completing properly. Although, come to think of it it might be a good idea to add a `"Bounty Hunting (X): done" ++` to all those bounty jobs.

I dont think we really need to keep track of it as we already have the combat rating for anything combat related

**Comment by mOctave** [comment]
I guess.
